beacon-chain/execution/engine_client.go

```diff
@@ -163,9 +163,20 @@ func (s *Service) NewPayload(ctx context.Context, payload interfaces.ExecutionDa
 		if !ok {
 			return nil, errors.New("execution data must be a Deneb execution payload")
 		}
-		err := s.rpcClient.CallContext(ctx, result, NewPayloadMethodV3, payloadPb, versionedHashes, parentBlockRoot)
-		if err != nil {
-			return nil, handleRPCError(err)
+		if executionRequests == nil {
```

Comment:
This feels prone to bugs and hard to test against . Checking payload's version may be better

Reply:
yeah , the caller checks the version on the block, but the payload doesn't include the version , I can pass the block version into this function if that makes it safer